synapse/storage/schema/delta/56/stats_separated1.sql

```diff
@@ -31,3 +31,118 @@ DELETE FROM background_updates WHERE update_name IN (
     'populate_stats_process_rooms',
     'populate_stats_cleanup'
 );
+
+----- Create tables for our version of room stats.
+
+-- single-row table to track position of incremental updates
+CREATE TABLE IF NOT EXISTS stats_incremental_position (
+    -- the stream_id of the last-processed state delta
+    state_delta_stream_id BIGINT,
+
+    -- the stream_ordering of the last-processed backfilled event
+    -- (this is negative)
+    total_events_min_stream_ordering BIGINT,
+
+    -- the stream_ordering of the last-processed normally-created event
+    -- (this is positive)
+    total_events_max_stream_ordering BIGINT,
+
+    -- If true, this represents the contract agreed upon by the background
+    -- population processor.
```

Comment:
What is the background population processor?

Reply:
The background process which performs initial counting of all room/users (initial generation or regeneration). It's had a few names over the time and I could do to unify it into one… but naming things is a hard problem. What do you think perhaps about 'stats regenerator'? (Pedantry might suggest it's not a great name since regeneration implies the stats already exist – but schmuh, I think this is probably my preferred choice at the end of thinking about it during this entire project)